include/rocksdb/sst_file_checksum.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+// Copyright (c) 2013 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#pragma once
+
+#include <cassert>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rocksdb/status.h"
+
+namespace rocksdb {
+
+// SstFileChecksumFunc is the function class to generates the checksum value
+// for each SST file when the file is written to the file system. The checksum
+// is stored in VersionSet as well as MANIFEST.
+class SstFileChecksumFunc {
```

Comment:
This is kind of nitpick: does this have to be specific to sst? Asking because file is named 'sst_file_checksum.h', but I think the implementation can work for any WritableFileWriter, correct?

Reply:
@riversand963 Yes, directly using FileChecksumFunc make scenes. Should I also change the file name as file_checksum.h?

Reply:
SG.